### ISSUE TITLE
Potential BUG in training wan with SP

### ISSUE THREAD

**Issue by zelingao98** [ByteDance-Seed/VeOmni#516]
Thanks for your amazing work!

In `train_wan.py`, I find the loss is not computed correctly. 

I think the right code should be corrected like this:

```bash
# shape: [B, ...], weight: [B]
loss = (loss.view(latents.size(0), -1).mean(dim=1) * weight).mean() / len(micro_batches)
# 👇 scale loss by sp_size
loss = loss / sp_size
```
Otherwise, for example, sp_size=2 means loss is computed twice during training as `wan_dit` will gather output. 

Please correct me if I am wrong.

**Comment by piyifan123** [comment]
Thanks for noting this.

@Coach257 is working on a major trainer refactoring and we can address this for the refactoring. @FoolPlayer cc

**Comment by Eisenhower** [comment]
Flagging for triage — this looks resolved by a subsequent architecture change rather than needing a direct patch.

`tasks/dit/train_wan.py` (referenced here) was moved to `tasks/deprecated_task/train_wan.py` in #570 (`support dit trainer & wan e2e training`) and is no longer the active Wan training entrypoint (`tasks/train_dit.py` is).

More importantly, the current Wan implementation (`veomni/models/diffusers/wan_t2v/wan_transformer/modeling_wan_transformer.py`) no longer computes the loss on an SP-local shard at all — it gathers the full sequence back on every SP rank *before* computing the loss (see the comment at `modeling_wan_transformer.py:232-235`: "every rank holds the full sequence so that the loss is identical across SP ranks"). That sidesteps the whole class of bug this issue describes (loss computed on a 1/sp_size shard but reduced as if it were the full batch) — there's no per-rank `sp_size` scaling left to get wrong, by construction.

Might be worth closing, unless you're hitting this on the old `deprecated_task/train_wan.py` path specifically.
